### PR TITLE
Jetpack: Sync when an attachment is added to a post for the first time

### PR DESCRIPTION
**Pull request by gititon**
This PR changes attachment update syncing to be triggered by `attachment_updated`, and syncs a specific action when the update changes the post_parent from 0 to non-zero, which is indicative of an attachment being attached to a post.

Testing Instructions:
Attach an existing post from your media library to a post by clicking `Add Media` in the post. You should see the `jetpack_sync_save_update_attachment` sync action come through on your wpcom sandbox.

phpunit tests were added to test this functionality as well

### CODE REVIEW

File: sync/class.jetpack-sync-module-attachments.php [hunk 1]
```diff
@@ -40,4 +42,19 @@ function send_attachment_info( $attachment_id ) {
...
 
 	}
+
+	function process_update( $attachment_id, $attachment_after, $attachment_before ) {
+		// Check whether attachment was added to a post for the first time
+		if ( 0 === $attachment_before->post_parent && 0 !== $attachment_after->post_parent ) {
+			/**
+			 * Fires when an existing attachment is added to a post for the first time
+			 *
+			 * @since 6.5.0
```
Comment (jeherve) [inline sync/class.jetpack-sync-module-attachments.php]: Since 6.5 is out, you'll need to update this to 6.6.0 I'm afraid!

Comment (roccotripaldi) [reply sync/class.jetpack-sync-module-attachments.php]: No prob, no big rush on this one!

File: sync/class.jetpack-sync-module-attachments.php [hunk 2]
```diff
@@ -6,38 +6,40 @@ function name() {
...
-			 *
-			 * @since 4.9.0
-			 *
-			 * @param int The attachment ID
-			 * @param object The attachment
-			 *
-			 * Previously this action was synced using jetpack_sync_save_add_attachment action.
-			 */
-			do_action( 'jetpack_sync_save_update_attachment', $attachment_id, $attachment );
+			do_action( 'jetpack_sync_save_update_attachment', $attachment_id, $attachment_after );
```
Comment (roccotripaldi) [inline sync/class.jetpack-sync-module-attachments.php]: We'll need to include the old docblock as well

**Comment by gititon** [comment]
@enejb @jeherve @roccotripaldi would you mind taking another look? Thanks!

File: tests/php/sync/test_class.jetpack-sync-posts.php [hunk 3]
```diff
@@ -261,6 +261,44 @@ public function test_sync_attachment_update_is_synced() {
...
+		$this->server_event_storage->reset();
+
+		wp_insert_attachment( $attachment, dirname( __FILE__ ) . '/../files/jetpack.jpg', 1000 );
+
+		$this->sender->do_sync();
+		
+		$attach_attachment_event = $this->server_event_storage->get_most_recent_event( 'jetpack_sync_save_attach_attachment' );
+		$update_attachment_event = $this->server_event_storage->get_most_recent_event( 'jetpack_sync_save_update_attachment' );
+		$add_attachment_event    = $this->server_event_storage->get_most_recent_event( 'jetpack_sync_save_add_attachment' );
+
```
Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-posts.php]: Lets also check that we have all the data that will make it so possible that the attachment is stored on the .com side as expected.

File: sync/class.jetpack-sync-module-attachments.php [hunk 4]
```diff
@@ -6,25 +6,38 @@ function name() {
...
+		 *
+		 * @param int The attachment ID
+		 * @param object The attachment
+		 */
+		do_action( 'jetpack_sync_save_add_attachment', $attachment_id, $attachment );
+	}
+
+	function process_update( $attachment_id, $attachment_after, $attachment_before ) {
+		// Check whether attachment was added to a post for the first time
+		if ( 0 === $attachment_before->post_parent && 0 !== $attachment_after->post_parent ) {
```
Comment (enejb) [inline sync/class.jetpack-sync-module-attachments.php]: Should we be checking here if the attachment has changed? 
Is it also possible for the attachment to be removed? 
Does that ever happen?

**Pull request merged**